### DISCUSSION TITLE
Tuya Plugin - unable to login

### DISCUSSION THREAD

**Discussion by Gonzoe79** [koush/scrypted#864] [Q&A]
Hi,

thanks for this nice project. i wanted to integrate my tuya ip cam (LSC Smart PTZ Camera), but i am am unable to login to tuya.
there is no other information in the logs that shows any clue whats wrong.

i followed the instructions and set everything up in tuya dev cloud, maybe there is something wrong with it

i set up a "Custom development" project  in tuya iot  cloud

for userId i use the "UID" of "Link App Account" in my Project in Tuya IOT Platform.
accessId and accessKey from the main project.

i chose Germany as Country (which should set the Datacenter URL to Central Europe)

 anyone has an idea what is wrong with the settings ? or are there currently issues with tuya ?
 
 best regards
 Gonzoe

**Comment by Gonzoe79** [comment]
just found out that i had to renew the subscription as mentioned in https://github.com/0x5e/homebridge-tuya-platform:

```
⚠️Remember to extend the API trial period every 6 months here [Tuya IoT Platform > Cloud > Cloud Services > IoT Core](https://iot.tuya.com/cloud/products/detail?abilityId=1442730014117204014&id=p1668587814138nv4h3n&abilityAuth=0&tab=1) (the first-time subscription only gives you 1 month).
```

turned out that my subscription had expired. i renewed it and  now i am waiting for approval by tuya.

i think it would be a good to mention it in the readme, since it really causes confusion.
also tuya seems to constantly rename / redesign their webpage and it might not be clear, where to get the credentials from.

**Comment by agborkowski** [comment]
wired I think you don't need a paid subscription for one IoT platform, in my case its works perfect i use same configuration like in HomeAssistant
